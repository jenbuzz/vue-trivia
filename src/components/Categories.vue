<template>
    <b-row align-h="center">
        <b-col sm="6" class="categories text-center">
            <button type="button" class="btn btn-category" @click="setCurrentCategory('art')">
                <img alt="Art Category" src="../assets/art.svg" />
                <span>Art</span>
            </button>
            <button
                type="button"
                class="btn btn-category"
                @click="setCurrentCategory('entertainment')"
            >
                <img alt="Entertainment Category" src="../assets/entertainment.svg" />
                <span>Entertainment</span>
            </button>
            <button type="button" class="btn btn-category" @click="setCurrentCategory('history')">
                <img alt="History Category" src="../assets/history.svg" />
                <span>History</span>
            </button>
        </b-col>
    </b-row>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class Categories extends Vue {
    @Prop() private setCurrentCategory!: Function;
}
</script>

<style scoped lang="scss">
.categories {
    flex-direction: row;
    display: flex;
    align-items: stretch;
    justify-content: center;

    .btn-category {
        display: flex;
        flex-direction: column;
        align-items: center;
        width: 140px;
        margin: 0 0.5rem;
        background: #e2e2e2;

        &:hover {
            background: #feb300;
        }

        img {
            width: 60px;
        }

        span {
            margin-top: auto;
            font-weight: bold;
        }
    }
}
</style>
