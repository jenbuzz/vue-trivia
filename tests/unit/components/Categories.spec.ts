import { createLocalVue, shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import Categories from '@/components/Categories.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('Component: Categories', () => {
    it('renders', () => {
        const wrapper = shallowMount(Categories, {
            localVue,
            propsData: { setCurrentCategory: jest.fn() },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
