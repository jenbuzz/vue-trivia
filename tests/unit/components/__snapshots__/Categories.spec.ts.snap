// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Categories renders 1`] = `
<b-row-stub tag="div" alignh="center">
  <b-col-stub sm="6" tag="div" class="categories text-center"><button type="button" class="btn btn-category"><img alt="Art Category" src="../assets/art.svg"> <span>Art</span></button> <button type="button" class="btn btn-category"><img alt="Entertainment Category" src="../assets/entertainment.svg"> <span>Entertainment</span></button> <button type="button" class="btn btn-category"><img alt="History Category" src="../assets/history.svg"> <span>History</span></button></b-col-stub>
</b-row-stub>
`;
